consensus/src/state_computer.rs

```diff
@@ -473,3 +481,149 @@ async fn test_commit_sync_race() {
     assert!(executor.sync_to(generate_li(2, 8)).await.is_ok());
     assert_eq!(*recorded_commit.time.lock(), LogicalTime::new(2, 8));
 }
+
+const PIPELINE_DEPTH: usize = 8;
+
+struct ExecutionPipeline {
+    block_tx: mpsc::Sender<ExecuteBlockCommand>,
+}
+
+impl ExecutionPipeline {
+    pub fn spawn(executor: Arc<dyn BlockExecutorTrait>, runtime: &tokio::runtime::Handle) -> Self {
+        let (block_tx, block_rx) = mpsc::channel(PIPELINE_DEPTH);
```

Comment:
What happens if we have more blocks in flight than the pipeline depth?

Reply:
line 513 will "block" (switch out) -- I think it's better than throwing in a lot of things unexpectedly into the pipeline.